src/server/utilities/constructCspHeader/index.test.js

```diff
@@ -0,0 +1,276 @@
+import injectCspHeader, {
+  generateScriptSrc,
+  generateImgSrc,
+  generateConnectSrc,
+} from './index';
+
+const next = jest.fn();
+
+const testAmpScript = (isAmp, isLive) => {
+  const expected = [
+    'https://cdn.ampproject.org',
+    'https://*.chartbeat.com',
+    'https://*.go-mpulse.net',
+    "'unsafe-inline'",
+  ];
+  const result = generateScriptSrc(isAmp, isLive);
+
+  expect(result).toEqual(expected);
+};
+
+const connectCommon = [
+  'https://*.akstat.io',
+  'https://*.akamaihd.net',
+  'https://c.go-mpulse.net',
+];
+
+const testConnect = (additional, isAmp, isUk, isLive) => {
+  const expected = connectCommon.concat(additional);
+  const result = generateConnectSrc(isAmp, isUk, isLive);
+
+  expect(result).toEqual(expected);
+};
+
+const testMiddleware = (origin, path, cspString) => {
+  const req = {
+    url: origin + path,
+    headers: {
+      'user-agent': 'local-agent',
+      'bbc-origin': origin,
+    },
+  };
+
+  const headers = {};
+
+  const res = {
+    setHeader: (key, value) => {
+      headers[key] = value;
+    },
+  };
+
+  injectCspHeader(req, res, next);
+
+  expect(next).toHaveBeenCalled();
+
+  expect(headers['Content-Security-Policy']).toEqual(cspString);
+};
+
+describe('Construct CSP Header', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should be able to generate the default live canon script src', async () => {
+    const expected = [
+      'https://news.files.bbci.co.uk',
+      'https://*.chartbeat.com',
+      'https://*.go-mpulse.net',
+      'https://mybbc-analytics.files.bbci.co.uk',
+      'https://emp.bbci.co.uk',
+      'https://static.bbci.co.uk',
+      "'self'",
+      "'unsafe-inline'",
+    ];
+    const result = generateScriptSrc(false, true);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should be able to generate the default test cannon script src', async () => {
+    const expected = [
+      'https://news.files.bbci.co.uk',
+      'https://*.chartbeat.com',
+      'https://*.go-mpulse.net',
+      'https://mybbc-analytics.files.bbci.co.uk',
+      'https://emp.bbci.co.uk',
+      'https://static.bbci.co.uk',
+      "'self'",
+      "'unsafe-inline'",
+      'https://news.test.files.bbci.co.uk',
+    ];
+    const result = generateScriptSrc(false, false);
+
+    expect(result).toEqual(expected);
+  });
+
+  it('should be able to generate the default live amp script src', async () => {
```

Comment:
lots of async functions here but are any of these tests actually async?

Reply:
Nope not really, I just followed a pattern that we had in the headers tests. I will remove the async keyword for readability then.